packages/aws-client/src/S3.ts

```diff
@@ -348,32 +348,58 @@ export const s3PutObjectTagging = improveStackTrace(
 );
 
 /**
-* Get an object from S3
-*
-* @param {string} Bucket - name of bucket
-* @param {string} Key - key for object (filepath + filename)
-* @param {Object} retryOptions - options to control retry behavior when an
-*   object does not exist. See https://github.com/tim-kos/node-retry#retryoperationoptions
-*   By default, retries will not be performed
-* @returns {Promise} returns response from `S3.getObject` as a promise
-**/
-export const getS3Object = improveStackTrace(
-  (Bucket: string, Key: string, retryOptions: pRetry.Options = { retries: 0 }) =>
-    pRetry(
-      async () => {
-        try {
-          return await s3().getObject({ Bucket, Key }).promise();
-        } catch (error) {
-          if (error.code === 'NoSuchKey') throw error;
-          throw new pRetry.AbortError(error);
+ * Gets an object from S3.
+ *
+ * @example
+ * const obj = await getObject(s3(), { Bucket: 'b', Key: 'k' })
+ *
+ * @param {AWS.S3} s3 - an `AWS.S3` instance
+ * @param {AWS.S3.GetObjectRequest} params - parameters object to pass through
+ *    to `AWS.S3.getObject()`
+ * @returns {Promise} response from `AWS.S3.getObject()` as a Promise
+ */
+// eslint-disable-next-line no-shadow
+export const getObject = (s3: AWS.S3, params: AWS.S3.GetObjectRequest) =>
```

Comment:
Is it not going to be a problem that we're deprecating `getS3Object` in favor of `getObject` when `getObject` has no retries? I suppose eventually the retries might not be as essential once we've updated everything to use etags when making requests, but there might be some interim where updating to use this function without retries could break things due to eventual consistency

Reply:
Yes, @yjpa7145 and I covered this. The idea is to separate concerns here so that the S3 functions are not concerned with retries. Callers will need to wrap retries around the functions themselves, if that's what's desired.

However, I do find use of `pRetries` a bit clumsy, and I had originally added a convenience function around the use of `pRetries` to make it simpler and more convenient to wrap retry behavior around any desired function, but pulled it out (before pushing to this PR) because I felt it would have raised further discussions that would have further delayed this PR.

Such a topic is perhaps worthy of discussion at the next Arch. Mtg.

Reply:
OK. I'm going to defer to @yjpa7145 on the refactored code